presto-main/src/main/java/io/prestosql/server/security/AuthenticationFilter.java

```diff
@@ -87,7 +87,9 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
                 response.setContentType(PLAIN_TEXT_UTF_8.toString());
                 return;
             }
-            nextFilter.doFilter(withPrincipal(request, principal), response);
+            withAuthenticatedIdentity(nextFilter, request, response, Identity.forUser("<internal>")
```

Comment:
I think we need another abstraction for `AuthenticatedIdentity`. Reusing `Identity` here looks like a hack, just see `"<internal>"` and not using roles and extra credentials.